xbmc/cores/VideoPlayer/VideoPlayer.h

```diff
@@ -530,4 +538,7 @@ class CVideoPlayer : public IPlayer, public CThread, public IVideoPlayer, public
   // omxplayer variables
   struct SOmxPlayerState m_OmxPlayerState;
   bool m_omxplayer_mode;            // using omxplayer acceleration
+
+  CCriticalSection m_player_status_lock;
```

Comment:
UpdateStreamInfos and GetVideoStreamInfo are called from different threads. Without the lock I might get inconsistent information (e.g. s.SrcRect may be partially updated).